pricelist_item_generator/models/pricelist.py

```diff
@@ -0,0 +1,415 @@
+                'auto': True}
+
+    @api.multi
+    def _unset_todo_create(self, tmpl_objs, condition_objs):
+        " Unset todo with 'create' value on both objects "
+        self.env['pricelist.item.template'].browse(
+            [x.id for x in tmpl_objs.keys()]).write({'todo': False})
+        self.env['pricelist.product.condition'].browse(
+            [x.id for x in condition_objs.keys()]).write({'todo': False})
+
+    @api.multi
+    def _synchronize_items(self, base_vals, vals):
+        self.ensure_one()
+        values = vals.copy()
+        values.update(base_vals)
+        values.update(self._complete_from_generator())
+        values['sequence'] = ((
+            int("{:0<4d}".format(self.sequence)) +
+            base_vals['sequence']))
+        return values
+
+    @api.multi
+    def _get_vals_from_objs(self, objects):
+        """ objects: iterable of object records of the models
+                    'pricelist.item.template' or 'pricelist.product.condition'
+            @return: list of dict of record values
+        """
+        self.ensure_one()
+        values_list = []
+        if objects and objects[0]:
+            fields = self._get_item_generator_fields(objects[0]._name)
+        for object in objects:
+            vals = {}
+            for field in fields:
+                if object[field]:
+                    vals[field] = object[field]
+                    if field == 'price_discount':
+                        # for usability purpose, discount is expressed
+                        # in a different way than in 'product.pricelist.item'
+                        # here, we convert the field
+                        vals[field] = - object[field] / 100
+                    if object._fields[field].type == 'many2one':
+                        vals[field] = object[field]['id']
+            if object._name == 'pricelist.item.template':
+                vals['item_template_id'] = object.id
+            else:
+                vals['product_condition_id'] = object.id
+            values_list.append(vals)
+        return values_list
+
+    @api.multi
+    def activate_generator(self):
+        for generator in self:
+            if generator.active is False:
+                vals = {'active': True, 'to_update': True}
+            else:
+                vals = {'active': False}
+                items_tpl = self.env['pricelist.item.template'].search(
+                    [('price_generator_id', '=', generator.id)])
+                if items_tpl:
+                    items2unlink = self.env['product.pricelist.item'].search([
+                        ('item_template_id', 'in', [x.id for x in items_tpl])])
+                    if items2unlink:
+                        items2unlink.unlink()
+                # 'todo' field to initial value
+                generator.item_template_ids.write({'todo': 'create'})
+                generator.product_condition_ids.write({'todo': 'create'})
+            generator.write(vals)
```

Comment:
Please group these writes into one using the Odoo ORM and the trick mentioned above. You would just want to make an array of `(1, record.id, {'values': 'dict'})` as you loop instead of directly writing.